lib/android/src/main/java/com/reactnativeldk/LdkModule.kt

```diff
@@ -653,6 +662,10 @@ class LdkModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaMod
                 return handleReject(promise, LdkErrors.channel_accept_fail, Error((error.err as APIError.APIMisuseError).err))
             }
 
+            if (error.err is APIError.ChannelUnavailable) {
+                return handleReject(promise, LdkErrors.channel_accept_fail, Error((error.err as APIError.ChannelUnavailable).err))
+            }
+
```

Comment:
Clearly all good on the logic side, though aesthetically I may have picked to replace the entire `if…` chaining with `when` because it enables handier [smart casting](https://arc.net/l/quote/fiepwmna):

```kt
            return when (val apiError = error.err) {
                is APIError.APIMisuseError -> {
                    handleReject(promise, LdkErrors.channel_accept_fail, Error((apiError).err))
                }

                is APIError.ChannelUnavailable -> {
                    handleReject(promise, LdkErrors.channel_accept_fail, Error((apiError).err))
                }

                else -> handleReject(
                    promise,
                    LdkErrors.channel_accept_fail,
                    Error(error.err.toString())
                )
            }
```

Reply:
Nice! Let me leave this to you you're welcome to comb through the kotlin side of things and cleanup any of these things you notice 💯